pkg/lib/packagekit.js

```diff
@@ -329,23 +324,35 @@ export function watchRedHatSubscription(callback) {
                 if (!contents || !yum_plugin_enabled_re.test(contents))
                     return;
 
-                // check if this is an unregistered RHEL system; if subscription-manager is not installed, ignore
-                var sm = cockpit.dbus("com.redhat.SubscriptionManager");
+                var sm = cockpit.dbus("com.redhat.RHSM1");
+
+                // check if this is an unregistered RHEL system; if
+                // subscription-manager is not installed, ignore
+
+                function check() {
+                    sm.call(
+                        "/com/redhat/RHSM1/Entitlement", "com.redhat.RHSM1.Entitlement", "GetStatus", ["", ""])
+                            .done(result => {
```

Comment:
Can you please use standard `.then`/`.done` JS promise API?

Reply:
Then! Err, Done! :-)